docs: revamp documentation with new structure and examples

guicara opened the pull request:
Hi,

This pull request introduces a major rewrite of the documentation for your awesome plugin.

**Key changes include:**

- Add a homepage
- Restructure dthe ocumentation file structure
- Add Mermaid diagram examples
- Provide usage examples for @panzoom/panzoom options
- Include selector usage examples

<img width="3456" height="2084" alt="image" src="https://github.com/user-attachments/assets/e67a96c5-205f-4b44-96d2-bb77f8f7fa95" />

<img width="3456" height="1916" alt="image" src="https://github.com/user-attachments/assets/4eae8ec2-a633-4371-96f3-a3d1a9d18077" />

<img width="3456" height="1748" alt="image" src="https://github.com/user-attachments/assets/2213e50a-dfad-434b-81bb-39276ee377ac" />

**Purpose of the change:**

The purpose of this change is to enhance the usability and clarity of the documentation, making it easier for users to understand and utilize the plugin effectively. The new structure and examples aim to improve the onboarding experience for developers.

**Contribution context:**

I appreciate the effort and dedication that has gone into forking and maintaining this plugin, and I would like to assist in making it even more accessible to the community. I plan to use this plugin in the internal Developer's Guide of the company I work for, [Audemars Piguet](https://www.audemarspiguet.com/). Thank you for maintaining this valuable plugin!

r74tech commented:
Hi, @guicara !

Thank you so much for this excellent documentation rewrite! The new structure is logical and the examples are comprehensive. This will definitely help users understand and utilize the plugin better.

I noticed a few minor things that could be improved:

1. Typos: "zoon" should be "zoom" in a couple of places:
  - `/docs/docs/04-examples/03-mermaid.md` line 3
  - `/docs/docs/03-getting-started/02-configuration.md` line 40 and line 70

2. Trailing spaces: It would be great if you could run a linter or formatter to clean these up.

3. Enhancement suggestion: For the `panzoom-exclude` example, it might be helpful to mention the `excludeClass` configuration option that enables this behavior.
This was my oversight in the original documentation, but it would be clearer if we could add something like:

```js title="docusaurus.config.js"
 module.exports = {
   themeConfig: {
     zoom: {
       // ... other options
       excludeClass: 'panzoom-exclude',
     }
   }
 };
```
This would make it clear that excludeClass is a configuration option, not just a CSS selector pattern.

Overall, this is a fantastic improvement to the documentation. Thank you for taking the time to create such a comprehensive guide!

guicara commented:
Hi @r74tech,

I'll fix 1. and 2.

For 3., the `excludeClass` option doesn't seem to work properly. Maybe I should create an issue?

With the following `zoom` object:

```js
zoom: {
  selectors: [
    'div.mermaid[data-processed="true"]',
    'div.docusaurus-mermaid-container',
    '.drawio',
    '.panzoom-example'
  ],
  wrap: true,
  timeout: 1000,
  excludeClass: 'panzoom-exclude',
},
```

And the `docs/docs/04-examples/03-mermaid.md` file:

`````md
## Pan and Zoom disabled

For smaller diagrams, you can disable pan and zoom functionality:

<div className="panzoom-exclude">

```mermaid
sequenceDiagram
    Alice->>John: Hello John, how are you?
    John-->>Alice: Great!
    Alice-)John: See you later!
```

</div>
`````

The zoom and pan functionality is still enabled:

<img width="1332" height="827" alt="image" src="https://github.com/user-attachments/assets/81d09546-320a-40c1-bdae-42bbf062ef54" />

guicara commented:
Hi @r74tech,

I’ve added Prettier and ESLint to the documentation website. Let me know if you’d like to have Prettier and ESLint added to the plugin as well (the easiest way would be to move the configuration and packages to the root directory).

Three new npm commands added:
- `format` for run Prettier and rewrite the files
- `lint` to run the linter (without changing the files)
- `lint:fix` to run and rewrite the files

I’ve also corrected the typo.

Best,

r74tech commented:
Thank you for the updates! I really appreciate your quick response to the feedback.

Regarding point 3 (the `excludeClass` configuration), you're right - it seems that functionality isn't currently implemented in the plugin. I apologize for the confusion. This would be a nice feature to have in the future.

If you'd like, feel free to create an issue for this enhancement - I'd be happy to collaborate on implementing it when time permits. Or if you're interested in adding this feature yourself, I'd be glad to review and merge it!

I'm approving this PR as the documentation improvements are excellent and will greatly benefit users.

Thank you again for your contribution and for helping make this plugin more accessible to the community!

r74tech commented:
@guicara 
Thank you for the updates! I really appreciate your quick response to the feedback.

Regarding point 3 (the `excludeClass` configuration), you're right - it seems that functionality isn't currently implemented in the plugin. I apologize for the confusion. This would be a nice feature to have in the future.

If you'd like, feel free to create an issue for this enhancement - I'd be happy to collaborate on implementing it when time permits. Or if you're interested in adding this feature yourself, I'd be glad to review and merge it!

I'm approving this PR as the documentation improvements are excellent and will greatly benefit users.

Thank you again for your contribution and for helping make this plugin more accessible to the community!

guicara commented:
@r74tech, you're welcome.

Thank you for merging this pull request.

How do you handle version bumping and the release of a new version?

> Regarding point 3 (the excludeClass configuration), you're right - it seems that functionality isn't currently implemented in the plugin. I apologize for the confusion. This would be a nice feature to have in the future.

> If you'd like, feel free to create an issue for this enhancement - I'd be happy to collaborate on implementing it when time permits. Or if you're interested in adding this feature yourself, I'd be glad to review and merge it!

I will open an issue and try to solve it.

For your information, I will also create a new pull request today to add a toolbar feature that allows the user to zoom in, zoom out and reset zoom. Similar to https://playg0n.github.io/mkdocs-panzoom/Usage/ and https://timmywil.com/panzoom/demo/.

r74tech commented:
@guicara
Thank you so much for the excellent PR!

Releases are automated through CI, so a new version will be published automatically after merging.

Looking forward to your toolbar feature PR (zoom in/out/reset)! I'll check it once you submit the PR.

Thanks for your contributions!
